Documentation/gettingstarted/dns.rst

```diff
@@ -0,0 +1,127 @@
+
+.. literalinclude:: ../../examples/policies/getting-started/dns-matchname.yaml
+
+Let's take a closer look at the policy: 
+
+* The first egress section uses ``toFQDNs: matchName`` specification to allow egress to ``api.twitter.com``. The destination DNS should match exactly the name specified in the rule. The ``endpointSelector`` allows only pods with labels ``class: mediabot`` to have the egress access.
+* The second egress section allows all pods in the ``default`` namespace to access ``kube-dns`` service. Note that ``rules: dns`` instructs Cilium to inspect and allow DNS lookups matching specified patterns. In this case, inspect and allow all DNS queries. 
+
+Let's apply the policy:
+
+.. parsed-literal::
+
+  $ kubectl create -f \ |SCM_WEB|\/examples/policies/getting-started/dns-matchname.yaml
+
+Testing the policy, we see that ``mediabot`` has access to ``api.twitter.com`` but doesn't have access to any other external service, for e.g., ``help.twitter.com``. And ``spaceship`` pod doesn't have access to twitter. 
+
+.. parsed-literal::
+
+    $ kubectl exec -it mediabot -- curl -sL https://api.twitter.com
+    ...
+    ...
+
+    $ kubectl exec -it mediabot -- curl -sL https://help.twitter.com
+    ^C 
+
+    $ kubectl exec -it spaceship -- curl -sL https://api.twitter.com
+    ^C  
+
+Step 5: DNS Policies Using Patterns 
+===================================
+
+The above policy controlled DNS access based on exact match of the DNS domain name. Often it is required to allow access to a subset of domains. Let's say, in the above example, ``mediabot`` pods need access to any Twitter sub-domain, i.e., the pattern ``*.twitter.com``. We can achieve this easily by changing the ``toFQDN`` rule to use ``matchPattern`` instead of ``matchName``.
+
+.. literalinclude:: ../../examples/policies/getting-started/dns-pattern.yaml
+
+.. parsed-literal::
+
+  $ kubectl apply -f \ |SCM_WEB|\/examples/policies/getting-started/dns-pattern.yaml
+
+Test that ``mediabot`` has access to multiple Twitter services for which the DNS matches the pattern ``*.twitter.com``. Important to note and test that this doesn't allow access to ``twitter.com`` because the ``*.`` in the pattern requires one subdomain to be present in the DNS name. You can simply add more ``matchName`` and ``matchPattern`` clauses to extend the access. 
+(See :ref:`DNS based` policies to learn more about specifying DNS rules using patterns and names.)
+
+.. parsed-literal:: 
+  
+   $ kubectl exec -it mediabot -- curl -sL https://help.twitter.com
+   ...
+
+   $ kubectl exec -it mediabot -- curl -sL https://about.twitter.com
+   ... 
+
+   $ kubectl exec -it mediabot -- curl -sL https://twitter.com
+   ^C 
+ 
+Step 6: Combining DNS and Port Rules
+====================================
+
+The DNS-based access can be restricted to a specific port by adding an L4 rules section. Continuing with the example, we will restrict access to Twitter services to port ``443``. 
```

Comment:
This is misleading. It can be combined with L7 as well. Even if you don't demonstrate L7 here you should world it in a way that it cannot be interpreted for DNS to be only combinable with L4.

Reply:
done